client/src/main/java/io/pravega/client/admin/impl/ReaderGroupManagerImpl.java

```diff
@@ -95,10 +97,23 @@ public void createReaderGroup(String groupName, ReaderGroupConfig config) {
                                               new ReaderGroupStateUpdatesSerializer(), new ReaderGroupStateInitSerializer(), SynchronizerConfig.builder().build());
         Map<SegmentWithRange, Long> segments = ReaderGroupImpl.getSegmentsForStreams(controller, config);
         synchronizer.initialize(new ReaderGroupState.ReaderGroupStateInit(config, segments, getEndSegmentsForStreams(config)));
+
+        if (config.isSubscriberForRetention()) {
+            Set<Stream> streams = config.getStartingStreamCuts().keySet();
+            streams.forEach(s -> getAndHandleExceptions(controller.addSubscriber(scope, s.getStreamName(), groupName), RuntimeException::new));
+        }
     }
 
     @Override
     public void deleteReaderGroup(String groupName) {
+        ReaderGroup group = getReaderGroup(groupName);
+        ReaderGroupConfig config = group.getReaderGroupConfig();
+
+        if (config.isSubscriberForRetention()) {
+            Set<Stream> streams = config.getStartingStreamCuts().keySet();
+            streams.forEach(s -> getAndHandleExceptions(controller.deleteSubscriber(scope, s.getStreamName(), groupName), RuntimeException::new));
```

Comment:
If the `controller.deleteSubscriber ` for one of the streams fails then `deleteSubscriber(..)` API remaining streams will not be executed. Is the user expected to retry?
Please update the javadocs for this API to convey the retry expections.

The same comment applies to the other APIs.